src/javascript/binary/websocket_pages/cashier/payment_agent_list.js

```diff
@@ -1,120 +1,114 @@
                 active     : false,
             });
         });
 
-        hiddenClass = 'hidden';
-        ddlCountriesID = '#target_country';
-        $paListContainer = $('#pa_list');
-        agentTemplate = $paListContainer.find('#accordion').html();
+        $pa_list_container = $('#pa_list');
+        $agent_template    = $pa_list_container.find('#accordion').html();
 
-        residence = Cookies.get('residence');
+        let residence = Cookies.get('residence');
         if (!residence || residence.length === 0) {
             residence = '00'; // just to get a list of payment agent Countries
         }
 
         sendRequest(residence, true);
     };
 
-    const sendRequest = function(country, isList) {
+    const sendRequest = (country, is_list) => {
         BinarySocket.send({
-            paymentagent_list: country || $(ddlCountriesID).val(),
-            passthrough      : isList ? { countries_list: '1' } : {},
+            paymentagent_list: country || $(ddl_countries_id).val(),
+        }).then((response) => {
+            if (is_list) {
+                populateCountriesList(response);
+            } else {
+                populateAgentsList(response.paymentagent_list.list);
+            }
         });
     };
 
-    const responseHandler = function(response) {
-        if (response.echo_req.passthrough && response.echo_req.passthrough.countries_list === '1') {
-            populateCountriesList(response);
-        } else {
-            populateAgentsList(response.paymentagent_list.list);
-        }
-    };
-
     // --------------------------
     // ----- Countries List -----
     // --------------------------
-    const populateCountriesList = function(response) {
-        const $ddlCountries = $(ddlCountriesID);
-        $ddlCountries.empty();
+    const populateCountriesList = (response) => {
+        const $ddl_countries = $(ddl_countries_id);
+        $ddl_countries.empty();
 
-        const cList = response.paymentagent_list.available_countries;
-        if (cList.length === 0) {
-            $ddlCountries.parent().addClass(hiddenClass);
+        const countries = response.paymentagent_list.available_countries;
+        if (countries.length === 0) {
+            $ddl_countries.parent().addClass(hidden_class);
             showEmptyListMsg();
             return;
         }
 
-        const requestedCountry = response.echo_req.paymentagent_list;
+        const requested_country = response.echo_req.paymentagent_list;
         let found = false;
-        cList.map(function(country) {
-            if (country === requestedCountry) {
+        countries.map((country) => {
+            if (country === requested_country) {
                 found = true;
             }
-            insertListOption($ddlCountries, country[1], country[0]);
+            insertListOption($ddl_countries, country[1], country[0]);
+            $('#target_country').removeClass(hidden_class);
+            $('.barspinner').addClass(hidden_class);
```

Comment:
Could we move these 2 lines to outside the loop, since we only need to add/remove class once?